ivy_tests/test_ivy/test_frontends/test_torch/test_pooling_functions.py

```diff
@@ -269,3 +269,42 @@ def test_torch_lp_pool1d(
         stride=stride[0],
         ceil_mode=False,
     )
+
+
+# avg_pool2d
+@handle_frontend_test(
+    fn_tree="torch.nn.functional.lp_pool2d",
+    dtype_x_k_s=helpers.arrays_for_pooling(
+        min_dims=4,
+        max_dims=4,
+        min_side=1,
+        max_side=4,
+    ),
+    norm_type=helpers.number(min_value=0.1, max_value=6),
+    test_with_out=st.just(False),
+)
+def test_torch_lp_pool2d(
+    dtype_x_k_s,
+    norm_type,
+    *,
+    test_flags,
+    frontend,
+    fn_tree,
+    on_device,
+):
+    input_dtype, x, kernel_size, stride, _ = dtype_x_k_s
+    # Torch ground truth func expects input to be consistent
+    # with a channels first format i.e. NCW
+    x[0] = x[0].reshape((x[0].shape[0], x[0].shape[-1], *x[0].shape[1:-1]))
```

Comment:
You should not be using `reshape` to permute the shape of a tensor, use `x[0].transpose(...)` instead (since x[0] is a numpy array).

Reply:
Out of curiosity, why is reshaping wrong here?

Reply:
Glad you asked actually, you can try this yourself
```python
import numpy as np

a = np.array([[1,2,3], [4,5,6]])
v1 = a.reshape(3, 2)
v2 = a.transpose(1, 0)
v1 == v2 # False
```
You can think of reshaping as flattening the array and then rearranging it in the specified requested shape, transposing or permuting just returns a different view of the same tensor, it does not change its internal memory structure or move things around.

```python
img[height][width][channel] == img.transpose(2, 0, 1)[channel][height][width] # True
img[height][width][channel] == img.reshape(channel_dims, height_dims, width_dims)[channel][height][width] # False
```